pkg/storegateway/series_chunks.go

```diff
@@ -417,6 +422,183 @@ func (c *loadingSeriesChunksSetIterator) Err() error {
 	return c.err
 }
 
+//nolint:unused // dead code while we are working on PR 3968
+type partialSeriesChunksSet struct {
```

Comment:
[nit] Please comment this struct and fields. It's not immediately clear what it is reading the code.